### PR TITLE
added instance group collection data source

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/d/is_instance_groups.html.markdown [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+layout: "ibm"
+page_title: "IBM : ibm_is_instance_groups"
+description: |-
+  Get information about InstanceGroupCollection
+subcategory: "VPC infrastructure"
+---
+
+# ibm_is_instance_groups
+
+Provides a read-only data source for InstanceGroupCollection. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax.
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_instance_groups.html.markdown:11]: Please add the link to document.

File: website/docs/d/is_instance_groups.html.markdown [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_instance_groups.html.markdown:12]: Please add the below note:
**Note:** 
VPC infrastructure services are a regional specific based endpoint, by default targets to `us-south`. Please make sure to target right region in the provider block as shown in the `provider.tf` file, if VPC service is created in region other than `us-south`.

**provider.tf**

```terraform
provider "ibm" {
  region = "eu-gb"
}
```

File: ibm/service/vpc/data_source_ibm_is_instance_groups_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	sshKeyName := fmt.Sprintf("testsshkey%d", randInt)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsInstanceGroupsDataSourceConfigBasic(vpcName, subnetName, sshKeyName, publicKey, templateName, instanceGroupName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_is_instance_groups.is_instance_groups", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_instance_groups.is_instance_groups", "instance_groups.#"),
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_instance_groups_test.go:35]: Please add the checks for all the mandatory fields.
https://pages.github.ibm.com/vpc/vpc-spec-artifacts/branch/master/public/2022-03-29.html#/Instance%20groups/list_instance_groups

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hi Deepak, 
Please find the review comments here.
Regards,
Sunitha

File: ibm/service/vpc/data_source_ibm_is_instance_groups_test.go (hunk 3 above)
Comment (SunithaGudisagarIBM) [reply ibm/service/vpc/data_source_ibm_is_instance_groups_test.go:35]: Hey also bring PR to up to date

**Comment by SunithaGudisagarIBM** [comment]
```
data "ibm_is_instance_groups" "is_instance_groups" {
}
```

```
sunitha@Sunithas-MacBook-Pro vpc-ibm-lb-config % terraform show
# data.ibm_is_instance_groups.is_instance_groups:
data "ibm_is_instance_groups" "is_instance_groups" {
    id              = "2022-07-25 07:36:01.16725 +0000 UTC"
    instance_groups = [
        {
            application_port   = 0
            created_at         = "2022-07-25T07:12:55.000Z"
            crn                = "crn:v1:staging:public:is:us-south:a/efe5afc483594adaa8325e2b4d1290df::instance-group:r134-deb71156-120d-4ef5-9d48-0ad9dd259779"
            href               = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instance_groups/r134-deb71156-120d-4ef5-9d48-0ad9dd259779"
            id                 = "r134-deb71156-120d-4ef5-9d48-0ad9dd259779"
            instance_template  = [
                {
                    crn     = "crn:v1:staging:public:is:us-south-1:a/efe5afc483594adaa8325e2b4d1290df::instance-template:0716-70bf0743-181f-42ca-99d8-121e7416881a"
                    deleted = []
                    href    = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instance/templates/0716-70bf0743-181f-42ca-99d8-121e7416881a"
                    id      = "0716-70bf0743-181f-42ca-99d8-121e7416881a"
                    name    = "sunitha-instance-temp"
                },
            ]
            load_balancer_pool = []
            managers           = []
            membership_count   = 1
            name               = "sunitha-instance-group"
            resource_group     = [
                {
                    href = "https://resource-controller.test.cloud.ibm.com/v2/resource_groups/11caaa983d9c4beb82690daab08717e9"
                    id   = "11caaa983d9c4beb82690daab08717e9"
                    name = "Default"
                },
            ]
            status             = "scaling"
            subnets            = [
                {
                    crn           = "crn:v1:staging:public:is:us-south-1:a/efe5afc483594adaa8325e2b4d1290df::subnet:0716-4b83a399-a222-4736-a0e0-44cd802a7e64"
                    deleted       = []
                    href          = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/subnets/0716-4b83a399-a222-4736-a0e0-44cd802a7e64"
                    id            = "0716-4b83a399-a222-4736-a0e0-44cd802a7e64"
                    name          = "cli-subnet-1"
                    resource_type = "subnet"
                },
            ]
            updated_at         = "2022-07-25T07:12:54.000Z"
            vpc                = [
                {
                    crn           = "crn:v1:staging:public:is:us-south:a/efe5afc483594adaa8325e2b4d1290df::vpc:r134-060fbfe6-4d0f-47b0-b9c5-94da8e719e22"
                    deleted       = []
                    href          = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/vpcs/r134-060fbfe6-4d0f-47b0-b9c5-94da8e719e22"
                    id            = "r134-060fbfe6-4d0f-47b0-b9c5-94da8e719e22"
                    name          = "cli-vpc-1"
                    resource_type = "vpc"
                },
            ]
        },
        {
            application_port   = 0
            created_at         = "2022-07-25T07:34:31.000Z"
            crn                = "crn:v1:staging:public:is:us-south:a/efe5afc483594adaa8325e2b4d1290df::instance-group:r134-c5282360-3fc5-41d4-9300-07381689a494"
            href               = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instance_groups/r134-c5282360-3fc5-41d4-9300-07381689a494"
            id                 = "r134-c5282360-3fc5-41d4-9300-07381689a494"
            instance_template  = [
                {
                    crn     = "crn:v1:staging:public:is:us-south-1:a/efe5afc483594adaa8325e2b4d1290df::instance-template:0716-065b8d03-3ece-4cbe-a92c-65a89484648b"
                    deleted = []
                    href    = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instance/templates/0716-065b8d03-3ece-4cbe-a92c-65a89484648b"
                    id      = "0716-065b8d03-3ece-4cbe-a92c-65a89484648b"
                    name    = "cli-tpl-1"
                },
            ]
            load_balancer_pool = []
            managers           = [
                {
                    deleted = []
                    href    = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instance_groups/r134-c5282360-3fc5-41d4-9300-07381689a494/managers/r134-6977f675-62e0-4bf3-b448-c6757089c4f3"
                    id      = "r134-6977f675-62e0-4bf3-b448-c6757089c4f3"
                    name    = "chase-watch-backlight-ambitious"
                },
            ]
            membership_count   = 1
            name               = "sunitha-testing"
            resource_group     = [
                {
                    href = "https://resource-controller.test.cloud.ibm.com/v2/resource_groups/11caaa983d9c4beb82690daab08717e9"
                    id   = "11caaa983d9c4beb82690daab08717e9"
                    name = "Default"
                },
            ]
            status             = "scaling"
            subnets            = [
                {
                    crn           = "crn:v1:staging:public:is:us-south-1:a/efe5afc483594adaa8325e2b4d1290df::subnet:0716-4b83a399-a222-4736-a0e0-44cd802a7e64"
                    deleted       = []
                    href          = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/subnets/0716-4b83a399-a222-4736-a0e0-44cd802a7e64"
                    id            = "0716-4b83a399-a222-4736-a0e0-44cd802a7e64"
                    name          = "cli-subnet-1"
                    resource_type = "subnet"
                },
            ]
            updated_at         = "2022-07-25T07:34:29.000Z"
            vpc                = [
                {
                    crn           = "crn:v1:staging:public:is:us-south:a/efe5afc483594adaa8325e2b4d1290df::vpc:r134-060fbfe6-4d0f-47b0-b9c5-94da8e719e22"
                    deleted       = []
                    href          = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/vpcs/r134-060fbfe6-4d0f-47b0-b9c5-94da8e719e22"
                    id            = "r134-060fbfe6-4d0f-47b0-b9c5-94da8e719e22"
                    name          = "cli-vpc-1"
                    resource_type = "vpc"
                },
            ]
        },
    ]
}
```

**Comment by deepaksibm** [comment]
moving to upstream

**Pull request closed without merging**